networking/eth2/src/test/java/tech/pegasys/teku/networking/eth2/rpc/beaconchain/methods/BeaconBlocksByRootMessageHandlerTest.java

```diff
@@ -106,4 +121,69 @@ private BeaconBlocksByRootRequestMessage createRequest(final List<SignedBeaconBl
 
     return blocks;
   }
+
+  @Test
+  public void validateResponse_phase0Spec_v1Request() {
+    final Optional<RpcException> result =
+        handler.validateResponse(V1_PROTOCOL_ID, chainBuilder.generateBlockAtSlot(5).getBlock());
+
+    assertThat(result).isEmpty();
+  }
+
+  @Test
+  public void validateResponse_phase0Spec_v2Request() {
+    final Optional<RpcException> result =
+        handler.validateResponse(V2_PROTOCOL_ID, chainBuilder.generateBlockAtSlot(5).getBlock());
+
+    assertThat(result).isEmpty();
+  }
+
+  @Test
+  public void validateResponse_altairSpec_v1RequestForPhase0Block() {
+    final Spec spec = TestSpecFactory.createMinimalWithAltairFork(UInt64.valueOf(32));
+    final BeaconBlocksByRootMessageHandler handler =
+        new BeaconBlocksByRootMessageHandler(spec, recentChainData);
+
+    final Optional<RpcException> result =
+        handler.validateResponse(V1_PROTOCOL_ID, chainBuilder.generateBlockAtSlot(5).getBlock());
+
+    assertThat(result).isEmpty();
+  }
+
+  @Test
+  public void validateResponse_altairSpec_v1RequestForAltairBlock() {
+    final Spec spec = TestSpecFactory.createMinimalWithAltairFork(UInt64.valueOf(32));
+    final BeaconBlocksByRootMessageHandler handler =
+        new BeaconBlocksByRootMessageHandler(spec, recentChainData);
+
+    final Optional<RpcException> result =
+        handler.validateResponse(V1_PROTOCOL_ID, chainBuilder.generateBlockAtSlot(32).getBlock());
+
+    assertThat(result)
+        .contains(new InvalidRpcMethodVersion("Must request altair blocks using v2 protocol"));
+  }
+
+  @Test
+  public void validateResponse_altairSpec_v2RequestForPhase0Block() {
+    final Spec spec = TestSpecFactory.createMinimalWithAltairFork(UInt64.valueOf(32));
+    final BeaconBlocksByRootMessageHandler handler =
+        new BeaconBlocksByRootMessageHandler(spec, recentChainData);
+
+    final Optional<RpcException> result =
+        handler.validateResponse(V2_PROTOCOL_ID, chainBuilder.generateBlockAtSlot(5).getBlock());
+
+    assertThat(result).isEmpty();
+  }
+
+  @Test
+  public void validateResponse_altairSpec_v2RequestForAltairBlock() {
```

Comment:
Should we have a test for requesting a mix of phase0 and Altair blocks using each version of the protocol?

Reply:
Couldn't hurt - added some tests and fixed and issue with the error handling